internal/providers/pluginfw/pluginfw.go

```diff
@@ -43,17 +44,27 @@ type DatabricksProviderPluginFramework struct {
 var _ provider.Provider = (*DatabricksProviderPluginFramework)(nil)
 
 func (p *DatabricksProviderPluginFramework) Resources(ctx context.Context) []func() resource.Resource {
-	return []func() resource.Resource{
-		qualitymonitor.ResourceQualityMonitor,
-		library.ResourceLibrary,
+	if os.Getenv("USE_SDK_V2") == "true" {
+		return []func() resource.Resource{}
+	} else {
+		return []func() resource.Resource{
+			qualitymonitor.ResourceQualityMonitor,
+			library.ResourceLibrary,
+		}
 	}
 }
 
 func (p *DatabricksProviderPluginFramework) DataSources(ctx context.Context) []func() datasource.DataSource {
-	return []func() datasource.DataSource{
-		cluster.DataSourceCluster,
-		volume.DataSourceVolumes,
-		registered_model.DataSourceRegisteredModel,
+	if os.Getenv("USE_SDK_V2") == "true" {
+		return []func() datasource.DataSource{
+			registered_model.DataSourceRegisteredModel,
+		}
+	} else {
+		return []func() datasource.DataSource{
+			cluster.DataSourceCluster,
+			volume.DataSourceVolumes,
+			registered_model.DataSourceRegisteredModel,
+		}
```

Comment:
Debatable decision - do we want to have more fine grained control such that we can use different environment variable values to control specific resources / data sources?

Reply:
As discussed, mentioning for visibility -- it would be better to have more finer grained control per resource so people can pass the resources they want to use sdkv2 for individually as otherwise they would have to use sdkv2 for all resources if there is a bug in one resource.